pulsar-broker/src/main/java/org/apache/pulsar/broker/web/WebService.java

```diff
@@ -266,6 +267,12 @@ public void addServlet(String path, ServletHolder servletHolder, boolean require
             attributeMap.forEach(context::setAttribute);
         }
         filterInitializer.addFilters(context, requiresAuthentication);
+
+        if (pulsar.getConfiguration().isHttpResponseCompressionEnabled()) {
+            GzipHandler gzipHandler = new GzipHandler();
+            context.setHandler(gzipHandler);
```

Comment:
They way I understand Jetty architecture, this shouldn't work. If you continue your test below and actually verify you have a minimum number of lines, and it has at least one metric you know for sure exists there, you will parse it and make sure it works - well, this test should fail, unless I don't understand Jetty.

Jetty works with handler nesting. You create a ServletContextHandler, and you wrap it with a GZipHandler.
If you call `gzipHandler.setHandler(servletHandler)`, it means you wrapped the servelet handler with GZip. Now you can add gzipHandler to the `handlers` list.

Here you seem to do the other way around. ServletHandler is wrapping gzipHandler.

Not sure if it should work.

So I would say: Please expand you test to make sure the context you get, can be uncompressed using GZip and it actually contain a valid expected Prometheus response.

Also, let's rename `context` to `servletHandler or `servletContextHandler` to know it's a handler. Context is confusing name.

Reply:
I checked the jetty document: https://eclipse.dev/jetty/documentation/jetty-9/index.html#gzip-filter and tested both usages. Both of them are works.

I add the unit test as you described.